higlass/server.py

```diff
@@ -256,48 +264,59 @@ def setup(self):
         disk_cache_size = 2 ** 25
         disk_cache_dir = self.diskcache_directory
         lru_capacity = 400
-        print(
-            "self.diskcache_directory",
-            self.diskcache_directory,
-            op.exists(self.diskcache_directory),
-        )
+        # print(
+        #     "self.diskcache_directory",
+        #     self.diskcache_directory,
+        #     op.exists(self.diskcache_directory),
+        # )
 
         def start_fuse(directory, protocol):
-            print("starting fuse")
-            fuse = FUSE(
-                HttpFs(
-                    protocol,
-                    disk_cache_size=disk_cache_size,
-                    disk_cache_dir=self.diskcache_directory,
-                    lru_capacity=lru_capacity,
-                ),
-                directory,
-                foreground=False,
-                allow_other=True
-            )
-        proc1 = mp.Process(target=start_fuse, args=[self.http_directory, 'http'])
+            # print("starting fuse")
+            try:
+                fuse = FUSE(
+                    HttpFs(
+                        protocol,
+                        disk_cache_size=disk_cache_size,
+                        disk_cache_dir=self.diskcache_directory,
+                        lru_capacity=lru_capacity,
+                    ),
+                    directory,
+                    foreground=False,
+                    allow_other=True
+                )
+            except RuntimeError as e:
+                if str(e) != "1":
+                    raise e
+
+        proc1 = mp.Process(
+            target=start_fuse, args=[self.http_directory, 'http']
+        )
         proc1.start()
         proc1.join()
 
-        proc2 = mp.Process(target=start_fuse, args=[self.https_directory, 'https'])
+        proc2 = mp.Process(
+            target=start_fuse, args=[self.https_directory, 'https']
+        )
         proc2.start()
         proc2.join()
 
     def teardown(self):
         try:
             if OS_NAME == 'Darwin':
+                sh.umount("HttpFs")
                 sh.umount("-l", self.http_directory)
             else:
                 sh.fusermount("-uz", self.http_directory)
-        except Exception as ex:
+        except Exception:
             pass
 
         try:
             if OS_NAME == 'Darwin':
+                sh.umount("HttpFs")
                 sh.umount("-l", self.https_directory)
```

Comment:
Does the -l parameter do anything? I thought we got rid of it?

Reply:
Good question. I looked up `man umount` and couldn't find any `-l` parameter. Besides, the `sh.umount("HttpFs")` is what worked for me on macOS `10.14`. Since I didn't know what `sh.umount("-l", self.https_directory)` is suppose to do I left it there.

Reply:
I just realized my branch was outdated somehow. The `-l` option is gone now